### PR TITLE
Introduce `Style` Protocol

### PR DESCRIPTION
**Pull request by JPToroDev**
# Introducing the `Style` Protocol
## Overview

This PR introduces a new `Style` protocol that enables type-safe, composable CSS styling for HTML elements. The implementation provides a declarative, SwiftUI-inspired approach to styling with first-class support for dynamic themes and environmental adaptations.

## Features

The `Style` protocol provides a foundation for building flexible, maintainable styling solutions:

- Type-safe style composition using `StyleBuilder`
- Scoped styling through optional namespace support
- Dynamic theme adaptation
- Chainable environmental conditions
- Material design support with automatic light/dark variants

## Implementation

The core protocol is defined as:

```swift
public protocol Style: Hashable, Equatable {
    associatedtype Body: Style
    static var prefix: String? { get }
    @StyleBuilder var body: Body { get }
    var className: String { get }
    static var id: String { get }
    var namespace: String { get }
}
```

### Basic Style Composition

Create dynamic styles that respond to color scheme changes:

```swift
struct MyTextStyle: Style {
    var body: some Style {
        LightMode("red") // Targets the color property by default
        DarkMode("blue")
    }
}

// Apply to text
Text("Dynamic color")
    .foregroundStyle(MyTextStyle())
```

### Property-Specific Styling

Target specific CSS properties:

```swift
struct MyBackgroundStyle: Style {
    var body: some Style {
        LightMode(.background, "red")
        DarkMode(.background, "blue")
    }
}

// Apply to background
Text("Dynamic color")
    .background(MyBackgroundStyle())
```

### Advanced Conditional Styling

Combine multiple environmental conditions:

```swift
// Blue in dark mode, red when the orientation is landscape
// AND the color scheme is light, the default color otherwise
struct MyBackgroundStyle: Style {
    var body: some Style {
        LightMode("red")
            .orientation(.landscape)
        DarkMode("blue")
    }
}
```

### Theme-Aware Components

Create components that adapt to theme changes:

```swift
struct CurrentThemeStyle: Style {
    var namespace: String
    var body: some Style {
        DarkMode("blue")
            .currentTheme(id: namespace)
        LightMode("red")
            .currentTheme(id: namespace)
    }
}

struct ThemeSwitcher: HTML {
    @Environment(\.themes) var themes
    var body: some HTML {
        ForEach(themes) { theme in
            Button(theme.name) {
                SwitchTheme(theme)
            }
            .foregroundStyle(CurrentThemeStyle(namespace: theme.name))
        }
    }
}
```

### `Material` Support

Implement SwiftUI-like adaptive material styles with automatic light/dark variants based on a custom theme's conformance to `LightTheme` and `DarkTheme`:

```swift
// Overrides material to use dark variant in a LightTheme
Text("Material")
    .background(.regularMaterial.colorScheme(.dark))
```

## Testing

- [ ] Protocol conformance tests
- [ ] Style builder composition tests
- [ ] Environmental condition tests
- [ ] Material design tests

## Documentation

- Added comprehensive API documentation
- Included usage examples for common scenarios
- Created migration guide for existing styling implementations

## Compatibility

This enhancement maintains backward compatibility with existing implementations while providing a path forward for more sophisticated styling solutions.

## Next Steps

- Complete test coverage
- Add performance benchmarks
- Create additional style presets

### CODE REVIEW

File: Sources/Ignite/Elements/VStack.swift [hunk 1]
```diff
@@ -20,7 +20,10 @@ public struct VStack: BlockHTML {
     public var columnWidth = ColumnWidth.automatic
 
     /// The vertical space between elements in pixels.
-    private var spacing: Int
+    private var spacing: Int?
+
+    /// The vertical space between elements by utility class.
+    private var spacingAmount: SpacingAmount?
```
Comment (twostraws) [inline Sources/Ignite/Elements/VStack.swift:26]: Having both `spacing` and `spacingAmount` isn't terribly clear. Could this be rethought?

File: Sources/Ignite/Modifiers/Hidden.swift [hunk 2]
```diff
@@ -10,25 +10,54 @@ struct HiddenModifier: HTMLModifier {
...
     /// Creates a new hidden modifier with a boolean flag
     /// - Parameter isHidden: Whether to hide the element
     init(isHidden: Bool = true) {
         self.isHidden = isHidden
+        self.mediaQuery = nil
+    }
+
+    /// Creates a new hidden modifier with a media query condition
+    /// - Parameter mediaQuery: The media query that triggers hiding
+    init(mediaQuery: any QueryType) {
```
Comment (twostraws) [inline Sources/Ignite/Modifiers/Hidden.swift:25]: Would this work better as a single initializer with two values?

File: Sources/Ignite/Types/ColorWeight.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// ColorWeight.swift
+// Ignite
+// https://www.github.com/twostraws/Ignite
+// See LICENSE for license information.
+//
+
+/// Constants that define the weight (lightness or darkness) of a color.
+public enum ColorWeight: Int {
+    /// 100: Extremely light variant, mixed with 80% white
+    case pale = 100
```
Comment (twostraws) [inline Sources/Ignite/Types/ColorWeight.swift:11]: This is another example of graduated words I find confusing – pale is lighter than lightest, and deep is darker than darkest.

Comment (JPToroDev) [reply Sources/Ignite/Types/ColorWeight.swift:11]: Good point—I'll rework these names.

File: Sources/Ignite/Elements/VStack.swift (hunk 1 above)
Comment (JPToroDev) [reply Sources/Ignite/Elements/VStack.swift:26]: Sure, I'll rework the names.

File: Sources/Ignite/Modifiers/Hidden.swift (hunk 2 above)
Comment (JPToroDev) [reply Sources/Ignite/Modifiers/Hidden.swift:25]: All set except for this.

**Pull request closed without merging**